perfkitbenchmarker/publisher.py

```diff
@@ -144,12 +143,23 @@ def AddMetadata(self, metadata, benchmark_spec):
 
       if vm.scratch_disks:
         data_disk = vm.scratch_disks[0]
+        # Legacy metadata keys
         metadata[name_prefix + 'scratch_disk_type'] = data_disk.disk_type
         metadata[name_prefix + 'scratch_disk_size'] = data_disk.disk_size
         metadata[name_prefix + 'num_striped_disks'] = (
             data_disk.num_striped_disks)
-        if data_disk.disk_type == disk.PIOPS:
+        if getattr(data_disk, 'iops', None) is not None:
           metadata[name_prefix + 'scratch_disk_iops'] = data_disk.iops
+          metadata[name_prefix + 'aws_provisioned_iops'] = data_disk.iops
+        # Modern metadata keys
+        metadata[name_prefix + 'data_disk_0_type'] = data_disk.disk_type
```

Comment:
We decided not to handle multiple disks right now but to label the disk metadata with `data_disk_0_` for future compatibility. This matches the current behavior.